### ISSUE TITLE
Make it possible for sqlite to be compiled so it can load extensions

### ISSUE THREAD

**Issue by jkitchin** [d12frosted/homebrew-emacs-plus#646]
I am using emacs-plus@30. It is great, thanks for maintaining these!

It has built in support for sqlite that works well except it can't load extensions.

When I run this,

```
(let ((db (sqlite-open)))
  (sqlite-execute db "select * from pragma_compile_options()"))
```
it shows `OMIT_LOAD_EXTENSION`, and Emacs does not have a function called `sqlite-load-extension` which the documentation suggests it should. I guess it is missing because sqlite was compiled without enabling, or explictly disabling extensions. It would be nice if this was enabled, or if an option to enable it like --with-load-sqlite-extensions was possible.

**Comment by jkitchin** [comment]
This doesn't seem to be an issue with this recipe, but the limited system sqlite. See https://emacs.stackexchange.com/questions/79956/how-do-you-load-extensions-in-sqlite-built-into-emacs?noredirect=1#comment134105_79956 for the solution I found.

**Comment by d12frosted** [comment]
I wonder. Maybe we could add sqlite as dependency, so you always get this?

**Comment by jkitchin** [comment]
The problem was not that I didn't have sqlite. The issue is that Mac has a system sqlite that does not allow loading extensions, and that is the one that Emacs builds with. It did not use the sqlite I had installed from Brew. The brew sqlite is keg-only so it doesn't interfere with the system one.

On one Mac I was able to force link it and then Emacs built with the sqlite i wanted. But on another Mac brew refuses to link it even with --force, so on that machine so far I am out of luck.
